draft-ietf-quic-transport.md

```diff
@@ -1607,22 +1606,30 @@ NOT subsequently advertise a smaller offset.  A sender may receive WINDOW_UPDATE
 frames out of order; a sender MUST therefore ignore any reductions in flow
 control credit.
 
+A receiver MUST close the connection with a
+QUIC_FLOW_CONTROL_RECEIVED_TOO_MUCH_DATA error ({{error-handling}}) if the
+peer violates the advertised stream or connection flow control windows.
+
 A sender MUST send BLOCKED frames to indicate it has data to write but is
 blocked by lack of connection or stream flow control credit.  BLOCKED frames are
 expected to be sent infrequently in common cases, but they are considered useful
 for debugging and monitoring purposes.
 
 A receiver advertises credit for a stream by sending a WINDOW_UPDATE frame with
-the StreamID set appropriately.  A receiver may simply use the current received
-offset to determine the flow control offset to be advertised.
+the StreamID set appropriately. A receiver may use the current offset of data
+consumed to determine the flow control offset to be advertised.
+A receiver MAY send copies of a WINDOW_UPDATE frame in multiple packets in order
+to make sure that the sender receives it even if one of the packets is lost. A
+sender MUST ignore duplicate WINDOW_UPDATE frames.
```

Comment:
It's an optimization. The WINDOW_UPDATE will get retransmitted and delivered **eventually** (depending on the details of the loss detection). A lost WINDOW_UPDATE means that the sender might get flow-control limited, either for one stream or even for the whole connection (if it was a connection-level WINDOW_UPDATE).

Clearly, this is an optimization, so I wrote that a sender MAY find a suitable algorithm. I just wanted to make sure that a client is required to ignore duplicate WINDOW_UPDATEs, since the only other statement about multiple WINDOW_UPDATEs is the following
>  A sender may receive WINDOW_UPDATE frames out of order; a sender MUST therefore ignore any reductions in flow control credit.

which is not explicit about duplicates.